packages/smartcontracts/src/tests/BridgeProxy.flushFund.spec.ts

```diff
@@ -0,0 +1,143 @@
+    const balance2ndReceiverAfter2ndFlush = await testToken.balanceOf(newFlushReceiveAddress);
+    expect(balance1stReceiverAfter2ndFlush).to.equal(balance1stReceiverBefore2ndFlush);
+    expect(balance2ndReceiverAfter2ndFlush).to.equal(balance2ndReceiverBefore2ndFlush.add(toWei('100')));
+  });
+
+  it('Should be able to change acceptableRemainingDays', async () => {
+    const { proxyBridge, flushReceiveSigner, testToken } = await loadFixture(deployContracts);
+    // latestResetTimestamp = block.timestamp + 59 = X + 59
+    await proxyBridge.addSupportedTokens(testToken.address, toWei('10'), (await time.latest()) + 60);
+    // block.timestamp = X + 1
+    await testToken.mint(proxyBridge.address, toWei('100'));
+    // increase time by 60 secs
+    // block.timestamp = X + 61
+    await time.increase(60);
+    // block.timstamp = X + 62 > X + 59 = latestResetTimestamp --> flush
+    const balanceBefore1stFlush = await testToken.balanceOf(flushReceiveSigner.address);
+    await proxyBridge.flushFund();
+    const balanceAfter1stFlush = await testToken.balanceOf(flushReceiveSigner.address);
+    expect(balanceAfter1stFlush).to.equal(balanceBefore1stFlush.add(toWei('80')));
+    await proxyBridge.changeAcceptableRemainingDays(4);
+    await testToken.mint(proxyBridge.address, toWei('100'));
+    const balanceBefore2ndFlush = await testToken.balanceOf(flushReceiveSigner.address);
+    await proxyBridge.flushFund();
+    const balanceAfter2ndFlush = await testToken.balanceOf(flushReceiveSigner.address);
+    expect(balanceAfter2ndFlush).to.equal(balanceBefore2ndFlush.add(toWei('80')));
+  });
+
+  it('Should not flush fund for a token if it is in the adding period', async () => {
+    const { proxyBridge, flushReceiveSigner, testToken } = await loadFixture(deployContracts);
+    // latestResetTimestamp = block.timestamp + 59 = X + 59
+    await proxyBridge.addSupportedTokens(testToken.address, toWei('10'), (await time.latest()) + 60);
+    // block.timestamp = X + 1
+    await testToken.mint(proxyBridge.address, toWei('100'));
+    const balanceBeforeFlush = await testToken.balanceOf(flushReceiveSigner.address);
+    // block.timestamp = X + 2
+    await proxyBridge.flushFund();
+    const balanceAfterFlush = await testToken.balanceOf(flushReceiveSigner.address);
+    expect(balanceAfterFlush).to.equal(balanceBeforeFlush);
+  });
+
+  it('Should not flush fund for a token if it is in the change allowance period', async () => {
+    const { proxyBridge, flushReceiveSigner, testToken, defaultAdminSigner } = await loadFixture(deployContracts);
+    // latestResetTimestamp = block.timestamp + 59 = X + 59
+    await proxyBridge.addSupportedTokens(testToken.address, toWei('10'), (await time.latest()) + 60);
+    const latestResetTimestampAfterAddingSupport = (await proxyBridge.tokenAllowances(testToken.address))
+      .latestResetTimestamp;
+    expect(latestResetTimestampAfterAddingSupport).to.equal((await time.latest()) + 59);
```

Comment:
```suggestion
    const expectedTime = getCurrentTimeStamp({ additionalTime: 60 });
    // latestResetTimestamp = block.timestamp + 59 = X + 59
    await proxyBridge.addSupportedTokens(testToken.address, toWei('10'), expectedTime);
    const latestResetTimestampAfterAddingSupport = (await proxyBridge.tokenAllowances(testToken.address))
      .latestResetTimestamp;
    expect(latestResetTimestampAfterAddingSupport).to.equal(expectedTime);
```

I'm guessing you were previously calling `getCurrentTimeStamp` multiple times, which was causing some time errors. If we did it in this manner with a constant reference point for the time, I don't think we'll have the problem of wrong timestamps (hardhat's `time.now()` vs node's `Date.now()`)

We can adjust the tests at the later date though - the logic still looks sound to me.

Reply:
Problem has been figured out. Opened https://github.com/WavesHQ/bridge/issues/357 to make sure that we use the fix approach as well for the other tests instead of the current `getCurrentTimeStamp` as well.